app/controllers/notes_controller.rb

```diff
@@ -55,10 +55,56 @@ def destroy
       track_destroyed(@note)
       redirect_to node_path(@node), notice: 'Note deleted'
     else
-      redirect_to node_note_path(@node, @note), alert: 'Could not delete node'
+      redirect_to node_note_path(@node, @note), alert: 'Could not delete note'
     end
   end
 
+  def multi_destroy
+    @notes = Note.where(
+      'node_id = :node AND id in (:ids)',
+      node: @node,
+      ids: params[:ids]
+    )
+
+    if @notes.any?
+      @job_logger = Log.new(uid: (Log.maximum(:uid) || 0) + 1)
+
+      if @notes.count > Note::MAX_DELETED_INLINE
+        @job_logger.write 'Enqueueing destroy job to start in the background.'
+        job = DestroyJob.perform_later(
```

Comment:
Maybe adding "Multi" would help to the name.

Also, I had to scroll down and see if this was a general AR multi-destroyer, but it seems it's tied to Note. So maybe we should include that in the name of the job too?

Reply:
My intention was to this be a general job, will recheck